File: tests/fixtures/cross_file_go_handler_exec/handler.go
package main

import (
	"fmt"
	"net/http"
)

// handleRun is an HTTP handler that takes a user-supplied command and executes
// it via a helper function defined in executor.go.
//
// VULN: r.FormValue("cmd") is a taint source (user-controlled).  The tainted
// value crosses a file boundary into runCommand(), which passes it verbatim to
// exec.Command.  No sanitisation occurs anywhere in the call chain.
func handleRun(w http.ResponseWriter, r *http.Request) {
	cmd := r.FormValue("cmd") // taint source: user input from HTTP form
	runCommand(cmd)            // tainted value crosses file boundary → exec.Command
	fmt.Fprintln(w, "done")
}

File: tests/fixtures/cross_file_py_object_field/handler.py
import subprocess

from models import JobRequest


def dispatch():
    """SINK: subprocess.call with shell=True.

    The command originates from JobRequest.cmd (an os.environ value) that was
    set in models.py.  The tainted attribute crosses the file boundary via the
    object reference and reaches the shell execution sink without sanitisation.

    VULN: object-field taint propagation from models.py into handler.py.
    """
    req = JobRequest()
    subprocess.call(req.cmd, shell=True)  # taint from req.cmd → shell sink
